crates/ink/codegen/src/generator/dispatch.rs

```diff
@@ -772,12 +788,10 @@ impl Dispatch<'_> {
 
                     push_contract(contract, #mutates_storage);
 
-                    if ::core::any::TypeId::of::<#message_output>() != ::core::any::TypeId::of::<()>() {
-                        // In case the return type is `()` we do not return a value.
-                        ::ink::env::return_value::<#message_output>(
-                            ::ink::env::ReturnFlags::default(), &result
-                        )
-                    }
+                    ::ink::env::return_value::<#message_output>(
+                        ::ink::env::ReturnFlags::default(), &result
+                    )
```

Comment:
And in fact it looks like this means that if a message returns `Result::Err` then that will now be the inner value so it would be `Ok(Err(err))`, which means that the contract will no longer be reverted and will be written to storage!